src/main/scala/nl/knaw/dans/bag/v0/DansV0Bag.scala

```diff
@@ -457,31 +458,52 @@ class DansV0Bag private(private[v0] val locBag: LocBag) extends DansBag {
    * @inheritdoc
    */
   override def addPayloadFile(inputStream: InputStream, pathInData: Path): Try[DansV0Bag] = Try {
-    val file: File = data / pathInData.toString
+    val dest: File = data / pathInData.toString
+    mustNotExistInBag(dest)
+    mustBeChildOfBagData(dest)
+    mustNotBeFetchFile(dest)
 
-    if (file.exists)
-      throw new FileAlreadyExistsException(file.toString)
-    if (!data.isParentOf(file))
-      throw new IllegalArgumentException(s"pathInData '$file' is supposed to point to a file that is a child of the bag/data directory")
-    if (fetchFiles.map(_.file) contains file)
-      throw new FileAlreadyExistsException(file.toString(), null, "file already present in bag as a fetch file")
+    dest.parent.createDirectories()
 
-    file.parent.createDirectories()
-
-    addFile(inputStream, file, locBag.getPayLoadManifests)
+    jFiles.copy(inputStream, dest)
+    addCheckSum(dest, locBag.getPayLoadManifests)
 
     this
   }
 
   /**
    * @inheritdoc
    */
-  override def addPayloadFile(src: File, pathInData: Path): Try[DansV0Bag] = Try {
-    addFile(src, pathInData)(bag => is => path => bag.addPayloadFile(is, path))
+  override def addPayloadFile(src: File, pathInData: Path)
+                             (implicit importOption: ImportOption): Try[DansV0Bag] = Try {
+    val dest = data / pathInData.toString
+    mustNotExistInBag(dest)
+    mustBeChildOfBagData(dest)
+    mustNotBeFetchFile(dest)
+    importOption match {
+      case ImportOption.COPY => addFile(src, pathInData)(_.addPayloadFile)
+      case ImportOption.MOVE => movePayloadFile(src, pathInData, atomicMove = false)
```

Comment:
@jo-pol , @rvanheest : What is the use case for non-atomic move?

Reply:
When you intend to move but allow the file system to fall back on copy+delete.